src/Xamarin.Android.Build.Tasks/Tasks/CompileToDalvik.cs

```diff
@@ -106,26 +113,31 @@ protected override string GenerateCommandLineCommands ()
 
 			cmd.AppendSwitch (DxExtraArguments);
 
+			cmd.AppendSwitchIfNotNull ("--input-list=", inputListFile);
+
 			if (MultiDexEnabled) {
 				cmd.AppendSwitch ("--multi-dex");
 				cmd.AppendSwitchIfNotNull ("--main-dex-list=", MultiDexMainDexListFile);
 			}
 			cmd.AppendSwitchIfNotNull ("--output ", Path.GetDirectoryName (ClassesOutputDirectory));
 
-
-			// .jar files
-			if (AlternativeJarFiles != null && AlternativeJarFiles.Any ()) {
-				Log.LogDebugMessage ("  processing AlternativeJarFiles...");
-				cmd.AppendFileNamesIfNotNull (AlternativeJarFiles, " ");
-			} else {
-				Log.LogDebugMessage ("  processing ClassesOutputDirectory...");
-				var zip = Path.GetFullPath (Path.Combine (ClassesOutputDirectory, "..", "classes.zip"));
-				if (!File.Exists (zip)) {
-					throw new FileNotFoundException ($"'{zip}' does not exist. Please rebuild the project.");
+			using (var sw = new StreamWriter (path: inputListFile, append: false,
+					encoding: new UTF8Encoding (encoderShouldEmitUTF8Identifier: false))) {
+				// .jar files
+				if (AlternativeJarFiles != null && AlternativeJarFiles.Any ()) {
+					Log.LogDebugMessage ("  processing AlternativeJarFiles...");
+					foreach (var l in AlternativeJarFiles)
+						sw.WriteLine (l.ItemSpec);
+				} else {
+					Log.LogDebugMessage ("  processing ClassesOutputDirectory...");
+					var zip = Path.GetFullPath (Path.Combine (ClassesOutputDirectory, "..", "classes.zip"));
+					if (!File.Exists (zip)) {
+						throw new FileNotFoundException ($"'{zip}' does not exist. Please rebuild the project.");
+					}
+					sw.WriteLine (zip);
+					foreach (var jar in JavaLibrariesToCompile)
+						sw.WriteLine (jar.ItemSpec);
```

Comment:
Ditto here: do we need to quote the paths?

Reply:
Weird this is what happens on Mac if I quote the files in the input list

```
/usr/bin/java -Xmx1G -jar /Users/dean/android-toolchain/sdk/build-tools/28.0.0/lib/dx.jar --dex --no-strict --input-list=/var/folders/n_/vt_nzq1n04schx5yy2jgcgcw0000gn/T/tmp799bbfa5.tmp --output obj/Debug/android/bin 
  
  UNEXPECTED TOP-LEVEL EXCEPTION:
  java.lang.RuntimeException: "/Users/dean/Documents/Sandbox/Xamarin/xamarin-android/bin/Debug/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v9.0/mono.android.jar": file not found
  	at com.android.dex.util.FileUtils.readFile(FileUtils.java:51)
  	at com.android.dx.cf.direct.ClassPathOpener.processOne(ClassPathOpener.java:168)
  	at com.android.dx.cf.direct.ClassPathOpener.process(ClassPathOpener.java:143)
  	at com.android.dx.command.dexer.Main.processOne(Main.java:678)
  	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:575)
  	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:310)
  	at com.android.dx.command.dexer.Main.runDx(Main.java:288)
  	at com.android.dx.command.dexer.Main.main(Main.java:244)
  	at com.android.dx.command.Main.main(Main.java:95)
```

This is the file

```
"/Users/dean/Documents/Sandbox/Xamarin/xamarin-android/bin/TestDebug/temp/BuildReleaseAppWithA InIt(FalseFalseFalse)/obj/Debug/android/bin/classes.zip"
"/Users/dean/Documents/Sandbox/Xamarin/xamarin-android/bin/Debug/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v9.0/mono.android.jar"
"/Users/dean/Documents/Sandbox/Xamarin/xamarin-android/bin/TestDebug/temp/BuildReleaseAppWithA InIt(FalseFalseFalse)/Hello (World).jar"
```

Reply:
So Mac does not like quotes... I will test on windows.